stablehlo/dialect/CMakeLists.txt

```diff
@@ -34,6 +34,11 @@ add_mlir_library(StablehloBase
   MLIRSupport
 )
 
+target_include_directories(StablehloBase INTERFACE
+  $<BUILD_INTERFACE:${STABLEHLO_SOURCE_DIR}>
+  $<BUILD_INTERFACE:${STABLEHLO_BINARY_DIR}>
+)
```

Comment:
From our conversation on #torch-mlir, I understand this is not an urgent PR, so let's spend a little bit more time iterating on it. I'd like to better understand the requirements for `target_include_directories`.
  * On the one hand, in MLIR-HLO we are also depending on StableHLO and have the same use case, but we're addressing this differently - by [using `include_directories`](https://github.com/tensorflow/mlir-hlo/blob/6a9a3ce04f8c01cc0c7bd7d574f339f73706149b/CMakeLists.txt#L137-L139) on the MLIR-HLO side.
  * IREE is doing something similar as well for their StableHLO dependency in CMake ([link](https://github.com/iree-org/iree/blob/094ec6d183e769893ae6e9ee98e668712a19dd24/build_tools/third_party/mlir-hlo/CMakeLists.txt#L22-L25)).
  * On the other hand, `target_include_directories` is not unprecedented in MLIR-HLO, e.g. [it's also used](https://github.com/tensorflow/mlir-hlo/blob/6a9a3ce04f8c01cc0c7bd7d574f339f73706149b/lib/Dialect/mhlo/IR/CMakeLists.txt#L52-L56) for the `MhloDialect` target.

@ashay Can you talk a bit more on the choice of targets - `StablehloBase` and `StablehloBroadcastUtils`? Do you anticipate you'd want `target_include_directories` for more targets in the future? Also what do you think about potentially using `include_directories` instead?